src/decode.c

```diff
@@ -1725,3 +1724,30 @@ bool rv_decode(struct rv_insn_t *ir, uint32_t insn, uint8_t *insn_len)
 #undef OP_UNIMP
 #undef OP
 }
+
+/* sign extend a 16 bit value */
```

Comment:
Move `sign_extend_h` and `sign_extend_b` to `riscv_private.h` since they are intended to be implementation specific.

Reply:
if move `sign_extend_h` and `sign_extend_b` to `riscv_private.h`, then `decode.c` needs to include `riscv_private.h`. It will make rv32emu compile failed because `riscv_private.h` includes the `gdbstub.h` as following and `gdbstub.h` is cloned until compiling `emulate.c`.
```c
#if RV32_HAS(GDBSTUB)
#include "breakpoint.h"
#include "mini-gdbstub/include/gdbstub.h"
#endif
```
Maybe, I think that `emulate.c` swaps the compiling order with `decode.c` will resolve this problem.
```diff
OBJS := \
	map.o \
	utils.o \
-       decode.o \
-       emulate.o \
+       emulate.o \
+       decode.o \
	riscv.o \
	io.o \
	elf.o \
	main.o \
	syscall.o \
	$(OBJS_EXT)
```

Reply:
It sounds fairly reasonable. Mention this within git commit messages.